OneCommunityGlobal/HighestGoodNetworkApp
# Rishitha material approval process

# Description
Implement Approval/Rejection process for the material purchase request.

## Related PRS (if any):
This frontend PR is related to the #[1138](https://github.com/OneCommunityGlobal/HGNRest/pull/1138) backend PR.
To test this frontend PR you need to checkout the #1138 backend PR.
…

## Main changes explained:
- Updated src/actions/bmdashboard/materialsActions.js 
- Updated src/components/BMDashboard/ItemList/ItemsTable.jsx to send purchace records correctly
- Updated src/components/BMDashboard/ItemList/RecordsModal.css to add styling to approve and reject buttons
- Updated src/components/BMDashboard/ItemList/RecordsModal.jsx to add approve and reject buttons and handling the buttons
- Updated src/constants/bmdashboard/materialsConstants.js
- Updated src/reducers/bmdashboard/materialsReducer.js
- Updated src/utils/URL.js to add backend endpoint
…

## How to test:
1. check into current branch 
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. Navigate to http://localhost:3000/bmdashboard and log in as an admin again.
6. Go to http://localhost:3000/bmdashboard/materials
7. Click on the View button under the purchases section to open the request details.
8. For each purchase request, verify that the Approve and Reject buttons are present and functional.
9. Click on the Approve button and ensure the request status updates correctly to "Approved".
10. Click on the Reject button and ensure the request status updates correctly to "Rejected".
11. Confirm that the total purchase count is updated accordingly with each action.
12. Ensure that the Approve and Reject buttons are disabled for requests that have already been approved or rejected.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/4bb32645-6c76-47bd-8d36-07a7556a36a3

## Note:
This is a phase 2 functionality.
